### ISSUE TITLE
QuickCheck not building

### ISSUE THREAD

**Issue by marnovandermaas** [CTSRD-CHERI/TestRIG#10]
I ran make on a vanilla checkout and got the following log. My system is Ubuntu 18.04 and using ghc 8.0.2:

```
$ make
cd vengines/QuickCheckVEngine &&\
if cabal --help 2>&1 | grep v1-build > /dev/null; then \
  (cabal v1-install --only-dependencies && cabal v1-configure && cabal v1-build); \
else \
  (cabal install --only-dependencies && cabal configure && cabal build); \
fi
Warning: The package list for 'hackage.haskell.org' is 229 days old.
Run 'cabal update' to get the latest list of available packages.
Resolving dependencies...
Downloading text-1.2.3.1...
Configuring text-1.2.3.1...
Building text-1.2.3.1...
Installed text-1.2.3.1
Downloading parsec-3.1.13.0...
Configuring parsec-3.1.13.0...
Building parsec-3.1.13.0...
Installed parsec-3.1.13.0
Resolving dependencies...
Configuring QCVEngine-0.1.0.0...
Building QCVEngine-0.1.0.0...
Preprocessing executable 'QCVEngine' for QCVEngine-0.1.0.0...
[ 1 of 41] Compiling RISCV.RV_CSRs    ( src/RISCV/RV_CSRs.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV_CSRs.o )
[ 2 of 41] Compiling RISCV.Helpers    ( src/RISCV/Helpers.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/Helpers.o )
[ 3 of 41] Compiling RISCV.ArchDesc   ( src/RISCV/ArchDesc.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/ArchDesc.o )
[ 4 of 41] Compiling QuickCheckVEngine.RVFI_DII.DII ( src/QuickCheckVEngine/RVFI_DII/DII.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/RVFI_DII/DII.o )

src/QuickCheckVEngine/RVFI_DII/DII.hs:86:13: error:
    • Variable not in scope: (<>) :: Put -> Put -> t1
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude), ‘*>’ (imported from Prelude),
        ‘<$’ (imported from Prelude)

src/QuickCheckVEngine/RVFI_DII/DII.hs:87:13: error:
    • Variable not in scope: (<>) :: t1 -> Put -> t0
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude), ‘*>’ (imported from Prelude),
        ‘<$’ (imported from Prelude)

src/QuickCheckVEngine/RVFI_DII/DII.hs:88:13: error:
    • Variable not in scope: (<>) :: t0 -> Put -> Put
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Prelude), ‘*>’ (imported from Prelude),
        ‘<$’ (imported from Prelude)
[ 5 of 41] Compiling InstrCodec       ( src/InstrCodec.hs, dist/build/QCVEngine/QCVEngine-tmp/InstrCodec.o )
[ 6 of 41] Compiling RISCV.RV32_A     ( src/RISCV/RV32_A.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_A.o )
[ 7 of 41] Compiling RISCV.RV32_D     ( src/RISCV/RV32_D.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_D.o )
[ 8 of 41] Compiling RISCV.RV32_F     ( src/RISCV/RV32_F.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_F.o )
[ 9 of 41] Compiling RISCV.RV32_I     ( src/RISCV/RV32_I.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_I.o )
[10 of 41] Compiling RISCV.RV32_M     ( src/RISCV/RV32_M.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_M.o )
[11 of 41] Compiling RISCV.RV32_Xcheri ( src/RISCV/RV32_Xcheri.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_Xcheri.o )
[12 of 41] Compiling RISCV.RV32_Zicsr ( src/RISCV/RV32_Zicsr.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_Zicsr.o )
[13 of 41] Compiling RISCV.RV32_Zifencei ( src/RISCV/RV32_Zifencei.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV32_Zifencei.o )
[14 of 41] Compiling RISCV.RV64_A     ( src/RISCV/RV64_A.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV64_A.o )
[15 of 41] Compiling RISCV.RV64_D     ( src/RISCV/RV64_D.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV64_D.o )
[16 of 41] Compiling RISCV.RV64_F     ( src/RISCV/RV64_F.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV64_F.o )
[17 of 41] Compiling RISCV.RV64_I     ( src/RISCV/RV64_I.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV64_I.o )
[18 of 41] Compiling RISCV.RV64_M     ( src/RISCV/RV64_M.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV64_M.o )
[19 of 41] Compiling RISCV.InstPretty ( src/RISCV/InstPretty.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/InstPretty.o )
[20 of 41] Compiling RISCV.RV_C       ( src/RISCV/RV_C.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV/RV_C.o )
[21 of 41] Compiling RISCV            ( src/RISCV.hs, dist/build/QCVEngine/QCVEngine-tmp/RISCV.o )
[22 of 41] Compiling QuickCheckVEngine.RVFI_DII.RVFI ( src/QuickCheckVEngine/RVFI_DII/RVFI.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/RVFI_DII/RVFI.o )
[23 of 41] Compiling QuickCheckVEngine.Template ( src/QuickCheckVEngine/Template.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Template.o )

src/QuickCheckVEngine/Template.hs:92:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘mappend’
    • In the instance declaration for ‘Monoid Template’

src/QuickCheckVEngine/Template.hs:136:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘mappend’
    • In the instance declaration for ‘Monoid TestCase’
[24 of 41] Compiling QuickCheckVEngine.Templates.Utils.FP ( src/QuickCheckVEngine/Templates/Utils/FP.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/Utils/FP.o )
[25 of 41] Compiling QuickCheckVEngine.Templates.Utils.General ( src/QuickCheckVEngine/Templates/Utils/General.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/Utils/General.o )
[26 of 41] Compiling QuickCheckVEngine.Templates.Utils.CHERI ( src/QuickCheckVEngine/Templates/Utils/CHERI.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/Utils/CHERI.o )
[27 of 41] Compiling QuickCheckVEngine.Templates.Utils.Compressed ( src/QuickCheckVEngine/Templates/Utils/Compressed.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/Utils/Compressed.o )
[28 of 41] Compiling QuickCheckVEngine.Templates.Utils ( src/QuickCheckVEngine/Templates/Utils.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/Utils.o )
[30 of 41] Compiling QuickCheckVEngine.Templates.GenAll ( src/QuickCheckVEngine/Templates/GenAll.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenAll.o )
[31 of 41] Compiling QuickCheckVEngine.Templates.GenArithmetic ( src/QuickCheckVEngine/Templates/GenArithmetic.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenArithmetic.o )
[32 of 41] Compiling QuickCheckVEngine.Templates.GenAtomics ( src/QuickCheckVEngine/Templates/GenAtomics.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenAtomics.o )
[33 of 41] Compiling QuickCheckVEngine.Templates.GenCHERI ( src/QuickCheckVEngine/Templates/GenCHERI.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenCHERI.o )
[34 of 41] Compiling QuickCheckVEngine.Templates.GenCSRs ( src/QuickCheckVEngine/Templates/GenCSRs.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenCSRs.o )
[35 of 41] Compiling QuickCheckVEngine.Templates.GenCompressed ( src/QuickCheckVEngine/Templates/GenCompressed.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenCompressed.o )
[36 of 41] Compiling QuickCheckVEngine.Templates.GenControlFlow ( src/QuickCheckVEngine/Templates/GenControlFlow.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenControlFlow.o )
[37 of 41] Compiling QuickCheckVEngine.Templates.GenFP ( src/QuickCheckVEngine/Templates/GenFP.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenFP.o )
[38 of 41] Compiling QuickCheckVEngine.Templates.GenMemory ( src/QuickCheckVEngine/Templates/GenMemory.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenMemory.o )
[39 of 41] Compiling QuickCheckVEngine.Templates.GenMulDiv ( src/QuickCheckVEngine/Templates/GenMulDiv.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/GenMulDiv.o )
[40 of 41] Compiling QuickCheckVEngine.Templates.RandomTest ( src/QuickCheckVEngine/Templates/RandomTest.hs, dist/build/QCVEngine/QCVEngine-tmp/QuickCheckVEngine/Templates/RandomTest.o )
Makefile:40: recipe for target 'QCVEngine' failed
make: *** [QCVEngine] Error 1
```

**Comment by marnovandermaas** [comment]
Fixed in commit: 477e16df628d0af9065578f552dbee37ff063974

**Comment by marnovandermaas** [comment]
Close
